test/layer.spec.ts

```diff
@@ -28,17 +33,179 @@ describe("applyLayers", () => {
+      extensionLayerVersion: EXTENSION_LAYER_VERSION,
+      apiKMSKey: "1234",
+      addLayers: true,
+      enableDDTracing: false,
+      flushMetricsToLogs: true,
+      site: "datadoghq.com",
+    });
+    datadogCdk.addLambdaFunctions([hello]);
+    expect(stack).toHaveResource("AWS::Lambda::Function", {
+      Layers: [
+        `arn:aws:lambda:${stack.region}:${DD_ACCOUNT_ID}:layer:Datadog-Python37:${PYTHON_LAYER_VERSION}`,
+        `arn:aws:lambda:${stack.region}:${DD_ACCOUNT_ID}:layer:Datadog-Extension:${EXTENSION_LAYER_VERSION}`,
+      ],
+    });
+  });
+
+  it("throws an error when the `extensionLayerVersion` is set and neither the `apiKey` nor `apiKMSKey` is set", () => {
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, "stack", {
+      env: {
+        region: "sa-east-1",
+      },
+    });
+    const hello = new lambda.Function(stack, "HelloHandler", {
+      runtime: lambda.Runtime.NODEJS_10_X,
+      code: lambda.Code.fromInline("test"),
+      handler: "hello.handler",
+    });
+    let threwError: boolean = false;
+    let thrownError: Error | undefined;
+    try{
+      const datadogCdk = new Datadog(stack, "Datadog", {
+        nodeLayerVersion: NODE_LAYER_VERSION,
+        extensionLayerVersion: EXTENSION_LAYER_VERSION,
+        addLayers: true,
+        enableDDTracing: false,
+        flushMetricsToLogs: true,
+        site: "datadoghq.com",
+      });
+      datadogCdk.addLambdaFunctions([hello]);
+    }catch(e){
+      threwError = true;
+      thrownError = e;
+    }
+    expect(threwError).toBe(true);
+    expect(thrownError?.message).toEqual("When `extensionLayer` is set, `apiKey` or `apiKMSKey` must also be set.");
+
+  });
+
+  it("throws an error when the `extensionLayerVersion` and `forwarderArn` are set", () => {
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, "stack", {
+      env: {
+        region: "sa-east-1",
+      },
+    });
+    const hello = new lambda.Function(stack, "HelloHandler", {
+      runtime: lambda.Runtime.NODEJS_10_X,
+      code: lambda.Code.fromInline("test"),
+      handler: "hello.handler",
+    });
+
+    let threwError: boolean = false;
+    let thrownError: Error | undefined;
+    try{
+      const datadogCdk = new Datadog(stack, "Datadog", {
+        nodeLayerVersion: NODE_LAYER_VERSION,
+        extensionLayerVersion: EXTENSION_LAYER_VERSION,
+        forwarderARN: "forwarder",
+        apiKey: "1234",
+        addLayers: true,
+        enableDDTracing: false,
+        flushMetricsToLogs: true,
+        site: "datadoghq.com",
+      });
+      datadogCdk.addLambdaFunctions([hello]);
+    }catch(e){
+      threwError = true;
+      thrownError = e;
+    }
+    expect(threwError).toBe(true);
+    expect(thrownError?.message).toEqual("`extensionLayerVersion` and `forwarderArn` cannot be set at the same time.");
+
+  });
+
+  it("throws an error when the `extensionLayerVersion` is set while `site` is undefined", () => {
```

Comment:
I am unsure if this unit test is appropriate because "site" is said to have a default value of datadoghq.com but the logic I have right now [here](https://github.com/DataDog/datadog-cdk-constructs/pull/21/files#diff-e07b5220c79ce96edc53faf651cde1714026204d4cfc4195eed15f516145fc8aR44) enforces that if the `extensionLayerVersion` is set then the customer needs to also set `site`.

Reply:
Ah that is a good point -- I think actually maybe we DON'T want to enforce that `site` must be set when `extensionLayerVersion` is set. Because, like you point out, we have a default value. So maybe we should remove that enforcement check. Unlike the API key, for which there obviously is no default, so they really do need to pass one in.

Reply:
I removed the enforcement of a site when extension layer version is provided, therefore a site is not required when the extension layer is provided however if they do provide it then it will be correctly added. I updated the readme accordingly

Reply:
That sounds good as long as we are defaulting to `datadoghq.com`.